### PR TITLE
Update abstract leveldown

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/Level/leveldown-hyper/issues/46

Precondition: https://github.com/Level/leveldown-hyper/pull/66 should be merged first

Squash this at merge.

### CODE REVIEW

File: test/approximate-size-test.js [hunk 1]
```diff
@@ -1,6 +1,122 @@
...
+        t.error(err)
+
+        db.approximateSize('!', '~', function (err, size) {
+          t.error(err)
+
+          t.equal(typeof size, 'number')
+          // account for snappy compression, original would be ~100000
+          t.ok(size > 40000, 'size reports a reasonable amount (' + size + ')')
+
+          db.close(function (err) {
```
Comment (vweevers) [inline test/approximate-size-test.js]: If I'm reading this right, the `db` here is the semi-global `var db` on line 5, which means the `db.close` in `tearDown` below is redundant?

Comment (vweevers) [reply test/approximate-size-test.js]: LGTM otherwise

Comment (ralphtheninja) [reply test/approximate-size-test.js]: > If I'm reading this right, the db here is the semi-global var db on line 5, which means the db.close in tearDown below is redundant?

Yep, seems like it. Same in `leveldown`.

File: test/approximate-size-test.js [hunk 2]
```diff
@@ -116,7 +116,3 @@ test('test approximateSize()', function (t) {
     })
   })
 })
-
-test('tearDown', function (t) {
-  db.close(testCommon.tearDown.bind(null, t))
```
Comment (vweevers) [inline test/approximate-size-test.js]: `testCommon.tearDown` should still be called though, right? To cleanup directories.

Comment (ralphtheninja) [reply test/approximate-size-test.js]: Indeed. Just noticed it will create the db under `node_modules/`

Comment (ralphtheninja) [reply test/approximate-size-test.js]: Removed the other `db.close()` instead :wink:

File: test/approximate-size-test.js [hunk 3]
```diff
@@ -106,11 +106,7 @@ test('test approximateSize()', function (t) {
           t.equal(typeof size, 'number')
           // account for snappy compression, original would be ~100000
           t.ok(size > 40000, 'size reports a reasonable amount (' + size + ')')
-
-          db.close(function (err) {
-            t.error(err)
-            t.end()
-          })
+          t.end()
```
Comment (vweevers) [inline test/approximate-size-test.js]: ah, clever :)

**Comment by ralphtheninja** [comment]
I'm wondering if we can skip the `apt-get` things on Travis, given that the compiler by default is good enough. We could get rid of these `apt-get` errors.

**Pull request merged**